### PR TITLE
Bump to k8s 1.25

### PR DESCRIPTION
**Pull request by a7i**
based on top of https://github.com/kubernetes-sigs/descheduler/pull/918

### CODE REVIEW

**Comment by damemi** [comment]
@a7i thanks for picking this up
Now that v0.25.0 is out, we can skip the rc.0 bump and go right to the GA (no point in soaking the rc when GA is already available)

**Comment by a7i** [comment]
Still waiting on this https://github.com/kubernetes/test-infra/pull/27348

**Comment by a7i** [comment]
😅 finally! let's get this reviewed/merged before something else breaks. I kept the commits separate to make the review easier but can squash if you suggest.

/cc @JaneLiuL @damemi @ingvagabund

**Review by JaneLiuL — COMMENTED** [review]
/lgtm
thanks @a7i

File: go.mod [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
-	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
+	k8s.io/api v0.25.0
+	k8s.io/apimachinery v0.25.0
+	k8s.io/apiserver v0.25.0
+	k8s.io/client-go v0.25.0
+	k8s.io/code-generator v0.25.0
+	k8s.io/component-base v0.25.0
+	k8s.io/component-helpers v0.25.0
+	k8s.io/klog/v2 v2.70.1
+	k8s.io/kubectl v0.25.0
```
Comment (damemi) [inline go.mod:17]: should we merge https://github.com/kubernetes-sigs/descheduler/pull/938 first, or after this?

Comment (damemi) [reply go.mod:17]: looks like #938 depends on this (specifically the go1.19 bump), so we can merge this first

**Pull request merged**